src/main/java/pl/ioad/adoto/database/TopObjectRepository.java

```diff
@@ -5,16 +5,43 @@
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 import pl.ioad.adoto.database.entity.Building;
+import pl.ioad.adoto.database.entity.River;
+import pl.ioad.adoto.database.entity.TopEntity;
 
 import java.util.List;
 
 
 @Repository
-public interface TopObjectRepository extends JpaRepository<Building, Long> {
-    @Query(value = "SELECT * FROM buildings " +
-            "WHERE ST_Intersects(wkb_geometry, ST_MakeEnvelope(:minX, :minY, :maxX, :maxY, 4326))",
-            nativeQuery = true)
-    List<Building> findIntersectingEntities(@Param("minX") double x1, @Param("minY") double y1,
-                                            @Param("maxX") double x2, @Param("maxY") double y2);
+public interface TopObjectRepository<T extends TopEntity> extends JpaRepository<T, Long> {
 
+    String INTERSECTING_BUILDINGS_QUERY = "SELECT * FROM buildings WHERE ST_Intersects(wkb_geometry, ST_MakeEnvelope(:minX, :minY, :maxX, :maxY, 4326))";
+    String INTERSECTING_RIVERS_QUERY = "SELECT * FROM rivers WHERE ST_Intersects(wkb_geometry, ST_MakeEnvelope(:minX, :minY, :maxX, :maxY, 4326))";
+    String INTERSECTING_ROADS_QUERY = "SELECT * FROM roads WHERE ST_Intersects(wkb_geometry, ST_MakeEnvelope(:minX, :minY, :maxX, :maxY, 4326))";
+    String INTERSECTING_FORESTS_QUERY = "SELECT * FROM forests WHERE ST_Intersects(wkb_geometry, ST_MakeEnvelope(:minX, :minY, :maxX, :maxY, 4326))";
+    String INTERSECTING_FIELDS_QUERY = "SELECT * FROM fields WHERE ST_Intersects(wkb_geometry, ST_MakeEnvelope(:minX, :minY, :maxX, :maxY, 4326))";
```

Comment:
consts should be static final and in this case private

Reply:
All fields in interface are implicitly `public static final`